app/controllers/granules_controller.rb

```diff
@@ -95,6 +95,63 @@ def each
     end
   end
 
+  def fetch_links
```

Comment:
This action is quite similar to ```download```. I had to steal a lot from ```download``` and ```GranuleUrlStreamer#each```. This ticket only addresses ```/granules/download.HTML```. ```/granules/download.sh``` still has the timeout issue if there are too many links to retrieve from CMR.

```EDSC-1124``` was created to address the timeout error caused by script file download. Once it is resolved,  the ```#download``` action will be much slimmer than it is now since most of the logic will be commonly used to generate the HTML page and the shell script file.